app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,39 @@
+class CoursesController < ApplicationController
```

Comment:
I think this should be in a Courses `module`, then in routes use `scope module: 'courses'`. That way we place all the Course modules in the Courses `module`, as submodules. So we have:
```ruby
module Courses; end
class Courses::CourseControllers < ApplicationController; end
class Courses::CoursesController < CourseControllers; end
module Courses::Assessments; end
class Courses::AssessmentControllers < CourseControllers; end
class Courses::Assessments::MissionsController < AssessmentControllers; end
...
...
module Courses::Forums; end
class Courses::ForumControllers < ApplicationController
class Courses::Forums::ForumsController < ForumControllers; end
...
```
This allows us to:
 - group controllers together by feature
 - implement breadcrumbs (as per bootstrap-sass-extras)

Hopefully the class hierarchy is then clearer.